### ISSUE TITLE
App closes immediately when trying to run tests on iOS Simulator

### ISSUE THREAD

**Issue by lucialuzuriaga** [webdriverio/webdriverio#3830]
I am constantly getting an ECONNREFUSED error when trying to run my tests on an iOS simulator .I reached a point in which, by the second WDA attempt, I wasn't getting the econnrefused error message, but the session was nevertheless interrupted/stopped immediately after launching the app.  

I have tried everything, including:
- Different iOS versions and device models
- Erasing content and settings, cleaning derived data folder
- Appium CLI and Desktop
- Different appium versions: 1.12.1, 1.9.0, 1.9.1
- Different desired capabilities (with or without reset, logs, bundle ID)
- I have chosen the same development team for both the app used and the WDA runner. 

The only issue that i haven’t been able to solve is a failure when running bootstrap script on WebDriverAgent folder. However when I open the project manually both lib and runner build successfully. I manually changed the bundle identifier to match the one of my app, but even if I leave it to the default com.facebook.WebDriverAgentRunner it still refuses to connect. 

I reported this same bug to Appium and according to them, there aren't any issues on their side according to the logs.
https://github.com/appium/appium/issues/12459

I have run out of ideas on why this is happening but it is frustrating and entirely blocking me from performing any work. 
The code is very simple, including only one login test case, following page object pattern as shown in https://github.com/webdriverio/appium-boilerplate

I am waiting for elements in the beforeEach loop, but I also tried adding browser.launch() or browser.reset() and I get the same results.
The only thing that comes to my mind is that the app has a splash screen at the beginning before showing the login. Could that by any chance have an impact?

- node version: 10.15.1
- npm version: 6.9.0
- OS: MacOS 10.14.4 Mojave
- Platform: iOS 
- Xcode Version: 10.1
- Appium version: 1.12.1 (also tried 1.9.0 and 1.9.1)
- Webdriverio CLI: 5.7.10

Package.json:
```
{
  "name": "automation-ios",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "android.sauce.rdc.app": "./node_modules/.bin/wdio ./config/saucelabs/wdio.android.rdc.app.conf.js",
    "android": "./node_modules/.bin/wdio ./config/wdio.android.app.conf.js",
    "android.browser": "./node_modules/.bin/wdio ./config/wdio.android.browser.conf.js",
    "ios": "./node_modules/.bin/wdio wdio.ios.conf.js",
    "ios.sauce.rdc.app": "./node_modules/.bin/wdio ./config/saucelabs/wdio.ios.rdc.app.conf.js",
    "ios.browser": "./node_modules/.bin/wdio ./config/wdio.ios.browser.conf.js",
    "lint": "eslint config tests"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@babel/cli": "^7.4.3",
    "@babel/core": "^7.4.3",
    "@babel/preset-env": "^7.4.3",
    "@babel/register": "^7.4.0",
    "@wdio/cli": "^5.7.10",
    "@wdio/jasmine-framework": "^5.7.8",
    "@wdio/local-runner": "^5.7.9",
    "@wdio/sync": "^5.7.9",
    "babel-preset-env": "^1.7.0",
    "babel-register": "^6.26.0",
    "wdio-appium-service": "^0.2.3"
  }
}
```

**My conf files:**
[sharedconf.txt](https://github.com/webdriverio/webdriverio/files/3059350/sharedconf.txt)
[iosconf.txt](https://github.com/webdriverio/webdriverio/files/3059448/iosconf.txt)

**Log**

Appium Logs with Xcode logs included:
[appiumlogs1.txt](https://github.com/webdriverio/webdriverio/files/3059455/appiumlogs1.txt)

Appium and iOS logs included:
[appiumios.txt](https://github.com/webdriverio/webdriverio/files/3059466/appiumios.txt)

**Comment by christian-bromann** [comment]
The logs look ok to me too .. I am not sure how to help you further

**Comment by christian-bromann** [comment]
@wswebcreation is the current boilerplate build broken?

**Comment by mgrybyk** [comment]
@lucialuzuriaga  not sure if it helps but you may try adding few more iOS capabilities 
```
    "useNewWDA": true,
    "waitForQuietness": false,
    "iosInstallPause": 2000,
```
also maybe running Appium itself with --session-override flag may help? I'll try to take a deeper look soon if nothing helps

**Comment by lucialuzuriaga** [comment]
@christian-bromann @mgrybyk I decided to make a new repository from scratch today and it worked without any code changes. However, the ECONNREFUSED error persisted even after getting the tests started running. I am under the idea that there is an underlying error behind this refused connection that is not being handled properly.

In case it's useful for someone reading this issue, my final capabilities were:
`config.capabilities = [
    {
        deviceName: 'iPhone 6',
        platformName: 'iOS',
        platformVersion: '12.1',
        orientation: 'PORTRAIT',
        maxInstances: 1,
        automationName: "XCUITest",
        xcodeOrgId: '123456789',
        xcodeSigningId: 'iPhone Developer',
        app: "path yo tour app",
        noReset: true,
        newCommandTimeout: 240,
    },
];`

**Comment by wswebcreation** [comment]
@lucialuzuriaga 

Can you provide an example project so we can check it. I run appium tests with WebdriverIO all the time and I don't see a problem, even the boilerplate project just works

**Comment by lucialuzuriaga** [comment]
> @lucialuzuriaga
> 
> Can you provide an example project so we can check it. I run appium tests with WebdriverIO all the time and I don't see a problem, even the boilerplate project just works

Maybe I should have clarified this, but I don't doubt that the boilerplate project works. This looks like an Appium problem. I searched the error thoroughly before posting, and it gets reported to their GitHub all the time with different frameworks, not just wdio. Since my logs looked fine they decided to close it and suggested that I reported it to Webdriverio. 

I cannot provide examples straight ahead due to privacy agreements with the client, I can provide one later after anonymizing a few variables. 
Not sure if this may help, but I checked the package.json files to see if there was any difference between both repositories, and what I've found was that I was using:
```
"@wdio/cli": "^5.7.10"
"@wdio/local-runner": "^5.7.9"
```
and now I'm using:
```
"@wdio/cli": "^5.7.12"
"@wdio/local-runner": "^5.7.12"
```
